bundler/spec/commands/binstubs_spec.rb

```diff
@@ -226,7 +226,10 @@
 
         it "calls through to the latest bundler version" do
           sys_exec "bin/bundle update --bundler", :env => { "DEBUG" => "1" }
-          expect(out).to include %(Using bundler #{system_bundler_version}\n)
+          word_found = /Using bundler ([\w\.]+)\n/.match(out)
```

Comment:
Looks good now! Only nitpick from my side is that I don't like the `word_found` name too much, I would use something a bit more descriptive like `using_bundler_line` or something. And maybe also using a named capture could make it more readable

```diff
diff --git a/bundler/spec/commands/binstubs_spec.rb b/bundler/spec/commands/binstubs_spec.rb
index aea897f03e..64aa4cbbc9 100644
--- a/bundler/spec/commands/binstubs_spec.rb
+++ b/bundler/spec/commands/binstubs_spec.rb
@@ -226,9 +226,9 @@
 
         it "calls through to the latest bundler version" do
           sys_exec "bin/bundle update --bundler", :env => { "DEBUG" => "1" }
-          word_found = /Using bundler ([\w\.]+)\n/.match(out)
-          expect(word_found).to_not be_nil
-          latest_version = word_found[1]
+          using_bundler_line = /Using bundler (?<version>[\w\.]+)\n/.match(out)
+          expect(using_bundler_line).to_not be_nil
+          latest_version = using_bundler_line[:version]
           expect(Gem::Version.new(latest_version)).to be >= Gem::Version.new(system_bundler_version)
         end
 
```

Reply:
OK. Let me fix it.

Reply:
I think RubyGems supports Ruby >= 2.3. seeing the `.rubocop.yml` file. Seeing Ruby 2.3 `Regexp` document, I didn't know `Regexp#match` returns a named capture, while `Regexp#named_captures` is explained on this purpose.
https://docs.ruby-lang.org/en/2.3.0/Regexp.html#method-i-match

Reply:
I did a quick check and RubyGems code base does not really use named captures at all at the moment, so let's just ignore that suggestion, sorry for the noise.

Reply:
Searching by `\.march`, I found 2 cases using maybe `String#match` https://docs.ruby-lang.org/en/2.3.0/String.html#method-i-match .

```
$ grep -r '\.match' bundler/spec/
...
bundler/spec/support/helpers.rb:            version = gem_name.match(/\Abundler-(?<version>.*)\z/)[:version] if gem_name != "bundler"
...
bundler/spec/support/builders.rb:        File.basename(file, ".gem").match(/\A(?<gem_name>[^-]+)-.*\z/)[:gem_name]
```

```
$ grep -r '\.named_captures' bundler/spec/
  => empty
```

The variable name `word_found` is used in some parts in the tests.

```
$ grep -r word_found bundler/spec/
bundler/spec/commands/binstubs_spec.rb:          word_found = /Using bundler ([\w\.]+)\n/.match(out)
bundler/spec/commands/binstubs_spec.rb:          expect(word_found).to_not be_nil
bundler/spec/commands/binstubs_spec.rb:          latest_version = word_found[1]
bundler/spec/quality_es_spec.rb:      next unless word_found = pattern.match(line)
bundler/spec/quality_es_spec.rb:      failing_line_message << "#{filename}:#{number.succ} contiene '#{word_found}'. Esta palabra tiene un significado subjetivo y es mejor obviarla en textos técnicos."
bundler/spec/quality_es_spec.rb:      next unless word_found = specific_pronouns.match(line)
bundler/spec/quality_es_spec.rb:      failing_line_message << "#{filename}:#{number.succ} contiene '#{word_found}'. Use pronombres más genéricos en la documentación."
bundler/spec/quality_spec.rb:      next unless word_found = pattern.match(line)
bundler/spec/quality_spec.rb:      failing_line_message << "#{filename}:#{number.succ} has '#{word_found}'. Avoid using these kinds of weak modifiers."
bundler/spec/quality_spec.rb:      next unless word_found = specific_pronouns.match(line)
bundler/spec/quality_spec.rb:      failing_line_message << "#{filename}:#{number.succ} has '#{word_found}'. Use more generic pronouns in documentation."
```

Considering this, if you suggest something, I am happy to consider modifying the PR's code.

Reply:
Well, `word_found` makes sense to me when regexp matching words, but in this case not so much. So my only suggestion would be to change that to a better variable name. The named captures is not necessary. As you point out, it's only used very scarcely in bundler's code.

Reply:
OK I will change `word_found` to `using_bundler_line`.

Reply:
No problem at all! I am happy on it.